core/src/test/java/org/apache/accumulo/core/file/FilePrefixTest.java

```diff
@@ -18,22 +18,79 @@
  */
 package org.apache.accumulo.core.file;
 
+import static org.apache.accumulo.core.file.FilePrefix.ALL;
+import static org.apache.accumulo.core.file.FilePrefix.BULK_IMPORT;
+import static org.apache.accumulo.core.file.FilePrefix.MAJOR_COMPACTION;
+import static org.apache.accumulo.core.file.FilePrefix.MAJOR_COMPACTION_ALL_FILES;
+import static org.apache.accumulo.core.file.FilePrefix.MERGING_MINOR_COMPACTION;
+import static org.apache.accumulo.core.file.FilePrefix.MINOR_COMPACTION;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
+import java.util.EnumSet;
+
 import org.junit.jupiter.api.Test;
 
 public class FilePrefixTest {
 
   @Test
-  public void testPrefixes() {
-    assertEquals(FilePrefix.BULK_IMPORT, FilePrefix.fromPrefix("I"));
-    assertEquals(FilePrefix.MINOR_COMPACTION, FilePrefix.fromPrefix("F"));
-    assertEquals(FilePrefix.MAJOR_COMPACTION, FilePrefix.fromPrefix("C"));
-    assertEquals(FilePrefix.MAJOR_COMPACTION_ALL_FILES, FilePrefix.fromPrefix("A"));
+  public void testFromPrefix() {
+    assertThrows(NullPointerException.class, () -> FilePrefix.fromPrefix(null));
+    assertThrows(IllegalArgumentException.class, () -> FilePrefix.fromPrefix(""));
+    assertThrows(IllegalArgumentException.class, () -> FilePrefix.fromPrefix("AB"));
+    assertEquals(MAJOR_COMPACTION, FilePrefix.fromPrefix("c"));
+    assertEquals(MAJOR_COMPACTION, FilePrefix.fromPrefix("C"));
```

Comment:
You could simplify similar test cases:

```suggestion
    Set.of("c", "C").forEach(s -> assertEquals(MAJOR_COMPACTION, FilePrefix.fromPrefix(s), "wrong type for prefix " + s));
```

Reply:
I can't implement this now that case sensitivity has been restored.